src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -963,6 +976,39 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
             return existing.WithAdditionalTreeOptions(builder.ToImmutable());
         }
 
+        private CompilerAnalyzerConfigOptionsProvider GetCompilerAnalyzerConfigOptionsProvider(
+            AnalyzerConfigSet? analyzerConfigSet,
+            ImmutableArray<AdditionalText> additionalTextFiles,
+            DiagnosticBag diagnostics,
+            ref Compilation compilation,
```

Comment:
It wasn't clear to me why we take `compilation` by reference here.

Reply:
The existing `CompileAndEmit()` method [has a `ref Compilation` parameter](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L972). I decided to follow the existing example and pass compilation by reference to this method as well.

Reply:
The `CompileAndEmit` method takes it by `ref` though because it can modify the value when attaching a compilation. Look at the `out compilation` usage for where this can change. This method does not modify the `compilation` though so shouldn't take it by `ref` here.

Reply:
I took the liberty of addressing this feedback.